### PR TITLE
How to upgrade kubernetes

### PR DESCRIPTION
**Pull request by iameskild**
## What does this implement/fix?

_Put a `x` in the boxes that apply_

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds a feature)
- [ ] Breaking change (fix or feature that would cause existing features not to work as expected)
- [x] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] Other (please describe):

## Testing

- [ ] Did you test the pull request locally?
- [ ] Did you add new tests?

#### Text styling

- [ ] The content is written with [plain language](https://www.plainlanguage.gov/guidelines/) (where relevant).
- [ ] If there are headers, they use the proper header tags (with only one level-one header: `H1` or `#` in markdown).
- [ ] All links describe where they link to (for example, check the [Nebari website](https://nebari.dev/)).
- [ ] This content adheres to the Nebari style guides.

#### Non-text content

- [ ] All content is represented as text (for example, images need alt text, and videos need captions or descriptive transcripts).
- [ ] If there are emojis, there are not more than three in a row.
- [ ] Don't use [flashing GIFs or videos](https://www.w3.org/TR/UNDERSTANDING-WCAG20/seizure-does-not-violate.html).
- [ ] If the content were to be read as plain text, it still makes sense, and no information is missing.

### CODE REVIEW

**Comment by iameskild** [comment]
@pavithraes @kcpevey these steps have been tested on each of the cloud providers and ready for review. Thanks :)

File: docs/docs/how-tos/upgrade-kubernetes-version.md [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+:::warning
+Downgrading to a lower version of Kubernetes is dangerous and we strongly advise against it!
+:::
+
+<Tabs>
+  
+<TabItem label="GCP" value="gcp" default="true">
+
+Google Kubernetes Engine (GKE) cut their own platform specific version of Kubernetes that usually look something like: `1.26.7-gke.500`; this corresponds to a Kubernetes version of `1.26.17`.
```
Comment (kenafoster) [inline docs/docs/how-tos/upgrade-kubernetes-version.md:41]: Is this a typo or does 1.26.7-gke.500 really correspond to 1.26.17?

File: docs/docs/how-tos/upgrade-kubernetes-version.md [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+</TabItem>
+
+
+<TabItem label="AWS" value="aws">
+
+The AWS Elastic Kubernetes Service (EKS) only requires that you supply the major and minor version of Kubernetes that you want. To specify Kubernetes version `1.26.7`, simply updated the `amazon_web_services.kubernetes_version` to `1.26`. Then run `nebari deploy` to apply these changes; you will get a validation error if you try to select a Kubernetes version that is unsupported by GKE or a version higher than `HIGHEST_SUPPORTED_K8S_VERSION`. This deployment process might take as long as 30 minutes.
+
+:::info
+You will get a validation error if you try to select a Kubernetes version that is unsupported by GKE or a version higher than [`HIGHEST_SUPPORTED_K8S_VERSION`](https://github.com/nebari-dev/nebari/blob/91792952b67074b5c15c3b4009bde5926ca4ec6b/src/_nebari/constants.py#L11).
```
Comment (kenafoster) [inline docs/docs/how-tos/upgrade-kubernetes-version.md:98]: "Unsupported by AWS"

File: docs/docs/how-tos/upgrade-kubernetes-version.md [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+:::
+
```
Comment (kenafoster) [inline docs/docs/how-tos/upgrade-kubernetes-version.md:100]: Blurb for node pools:

In AWS, upgrading EKS will upgrade the control plane components 

In the AWS console, navigate to EKS and click on the name of your Kubernetes cluster (format will be {{project-name}}-{{namespace}}.  In the 'Compute' tab, scroll down to "Node Groups".  Any node groups which are behind will have an "Update Now" button by the "AMI release version" column values.  Click "Update Now" for each.  Each update may take 15 or more minutes depending on how many workloads need to be migrated, but they can be run simultaneously."

File: docs/docs/how-tos/upgrade-kubernetes-version.md (hunk 1 above)
Comment (iameskild) [reply docs/docs/how-tos/upgrade-kubernetes-version.md:41]: Typo :)

File: docs/docs/how-tos/upgrade-kubernetes-version.md (hunk 3 above)
Comment (iameskild) [reply docs/docs/how-tos/upgrade-kubernetes-version.md:104]: Thanks @kenafoster :)

File: docs/docs/how-tos/nebari-upgrade.md [hunk 4]
```diff
@@ -42,6 +42,10 @@ nebari upgrade -c nebari-config.yaml
 
 The upgrade command may print some manual instructions that need to be run before continuing with the upgrade. If so, complete those instructions before continuing on.
 
+:::info
+New versions of Nebari often include support for newer versions of Kubernetes, as such you might need to upgrade your cluster's Kubernetes version. Please refer to [these docs](./upgrade-kubernetes-version.md) for more details.
```
Comment (pavithraes) [inline docs/docs/how-tos/nebari-upgrade.md:46]:
```suggestion
New versions of Nebari often include support for newer versions of Kubernetes, as such you might need to upgrade your cluster's Kubernetes version. Please refer to [Upgrade Kubernetes version docs](./upgrade-kubernetes-version.md) for more details.
```

File: docs/docs/how-tos/upgrade-kubernetes-version.md [hunk 5]
```diff
@@ -0,0 +1,159 @@
+---
+id: kubernetes-version-upgrade
+title: How to upgrade the cluster's Kubernetes version
```
Comment (pavithraes) [inline docs/docs/how-tos/upgrade-kubernetes-version.md:3]:
```suggestion
title: Upgrade cluster's Kubernetes version
```

File: docs/docs/how-tos/upgrade-kubernetes-version.md [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+:::warning
+Upgrading the kubernetes version of the node groups will cause downtime so please plan accordingly. We also recommend [backing up your data](./manual-backup.md) before starting this upgrade process.
+:::
+
+:::warning
+Kubernetes versions can only be upgraded one minor version at a time. If you're running on 1.24, and need to upgrade to 1.26, you will first need to upgrade to 1.25.
+:::
+
+:::warning
+Downgrading to a lower version of Kubernetes is dangerous and we strongly advise against it!
```
Comment (pavithraes) [inline docs/docs/how-tos/upgrade-kubernetes-version.md:34]:
```suggestion
1. Upgrading the kubernetes version of the node groups will cause downtime so please plan accordingly. We also recommend [backing up your data](./manual-backup.md) before starting this upgrade process.
2. Kubernetes versions can only be upgraded one minor version at a time. If you're running on 1.24, and need to upgrade to 1.26, you will first need to upgrade to 1.25.
3. Downgrading to a lower version of Kubernetes is dangerous and we strongly advise against it!
```

File: docs/docs/how-tos/upgrade-kubernetes-version.md [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+title: How to upgrade the cluster's Kubernetes version
+description: A basic overview on how to upgrade your cluster Kubernetes version
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+## Overview
+
+Nebari runs on Kubernetes under the hood, and as administrators of this Kubernets cluster, one of the maintenance tasks required from time to time includes upgrading the version of your Kubernetes cluster. Each of the different cloud providers that Nebari can run on, has their own release cycle and support windows for their flavor of Kubernetes. That said, they tend to follow the [official Kubernetes release cycle](https://kubernetes.io/releases/).
```
Comment (pavithraes) [inline docs/docs/how-tos/upgrade-kubernetes-version.md:12]:
```suggestion
Nebari runs on Kubernetes under the hood, and as administrators of this Kubernetes cluster, one of the maintenance tasks required from time to time is upgrading the version of your Kubernetes cluster. Each of the different cloud providers that Nebari can run on, has their own release cycle and support windows for their flavor of Kubernetes. That said, they tend to follow the [official Kubernetes release cycle](https://kubernetes.io/releases/).
```

File: docs/docs/how-tos/upgrade-kubernetes-version.md [hunk 8]
```diff
@@ -0,0 +1,159 @@
...
+
+The Nebari development team tries to stay ahead of this by supporting the latest version when possible. However given that many Kubernetes releases comes with a set of deprecations that potentially affect Nebari and downstream plugins, there is an enforced [`HIGHEST_SUPPORTED_K8S_VERSION`](https://github.com/nebari-dev/nebari/blob/91792952b67074b5c15c3b4009bde5926ca4ec6b/src/_nebari/constants.py#L11) allowed.
```
Comment (pavithraes) [inline docs/docs/how-tos/upgrade-kubernetes-version.md:14]:
```suggestion
The Nebari development team tries to stay ahead of this by supporting the latest version when possible. However, given that many Kubernetes releases come with a set of deprecations that potentially affect Nebari and downstream plugins, there is an enforced [`HIGHEST_SUPPORTED_K8S_VERSION`](https://github.com/nebari-dev/nebari/blob/91792952b67074b5c15c3b4009bde5926ca4ec6b/src/_nebari/constants.py#L11) allowed.
```

File: docs/docs/how-tos/upgrade-kubernetes-version.md [hunk 9]
```diff
@@ -0,0 +1,159 @@
...
+
+
+:::note
+This `HIGHEST_SUPPORTED_K8S_VERSION` is, at times, a minor version or two behind the officially released Kubernetes version.
+:::
+
+Many cloud providers enable users to automatically upgrade their Kubernetes cluster (control plane), however given the potential for deprecations and other changes, Nebari deployed to these cloud have this feature automatically disabled.
```
Comment (pavithraes) [inline docs/docs/how-tos/upgrade-kubernetes-version.md:21]:
```suggestion
Many cloud providers enable users to automatically upgrade their Kubernetes cluster (control plane), however given the potential for deprecations and other changes, Nebari deployed to these clouds has this feature automatically disabled. 
```

File: docs/docs/how-tos/upgrade-kubernetes-version.md [hunk 10]
```diff
@@ -0,0 +1,159 @@
...
+```
+
+For more information about GKE upgrades, please refer to the [GKE documentation](https://cloud.google.com/kubernetes-engine/docs/how-to/upgrading-a-cluster).
+
+</TabItem>
+
+
+<TabItem label="AWS" value="aws">
+
+The AWS Elastic Kubernetes Service (EKS) only requires that you supply the major and minor version of Kubernetes that you want. To specify Kubernetes version `1.26.7`, simply updated the `amazon_web_services.kubernetes_version` to `1.26`. Then run `nebari deploy` to apply these changes; you will get a validation error if you try to select a Kubernetes version that is unsupported by EKS or a version higher than `HIGHEST_SUPPORTED_K8S_VERSION`. This deployment process might take as long as 30 minutes.
```
Comment (pavithraes) [inline docs/docs/how-tos/upgrade-kubernetes-version.md:95]:
```suggestion
The AWS Elastic Kubernetes Service (EKS) only requires that you supply the major and minor version of Kubernetes that you want. To specify Kubernetes version `1.26.7`, update the `amazon_web_services.kubernetes_version` to `1.26`. Then run `nebari deploy` to apply these changes. This deployment process might take as long as 30 minutes.
```

File: docs/docs/how-tos/upgrade-kubernetes-version.md [hunk 11]
```diff
@@ -0,0 +1,159 @@
...
+
+:::info
+You will get a validation error if you try to select a Kubernetes version that is unsupported by EKS or a version higher than [`HIGHEST_SUPPORTED_K8S_VERSION`](https://github.com/nebari-dev/nebari/blob/91792952b67074b5c15c3b4009bde5926ca4ec6b/src/_nebari/constants.py#L11).
+:::
+
+In AWS, upgrading EKS will upgrade the control plane components but the node groups will need to be **upgraded manually**.
```
Comment (pavithraes) [inline docs/docs/how-tos/upgrade-kubernetes-version.md:101]:
```suggestion
In AWS, upgrading EKS will upgrade the control plane components but the **node groups will need to be upgraded manually**.
```

File: docs/docs/how-tos/upgrade-kubernetes-version.md [hunk 12]
```diff
@@ -0,0 +1,159 @@
...
+
+In the AWS console, navigate to EKS and click on the name of your Kubernetes cluster (format will be `{project-name}-{namespace}`). In the 'Compute' tab, scroll down to "Node Groups". Any node groups which are behind will have an "Update Now" button by the "AMI release version" column values. Click "Update Now" for each. Each update may take 15 or more minutes depending on how many workloads need to be migrated, but they can be run simultaneously."
```
Comment (pavithraes) [inline docs/docs/how-tos/upgrade-kubernetes-version.md:103]:
```suggestion
In the AWS console, navigate to EKS and click on the name of your Kubernetes cluster (format will be `{project-name}-{namespace}`). In the 'Compute' tab, scroll down to "Node Groups". Any node groups which are behind will have an "Update Now" button by the "AMI release version" column values. Click "Update Now" for each. Each update may take 15 or more minutes depending on how many workloads need to be migrated, but they can be run simultaneously.
```

File: docs/docs/how-tos/upgrade-kubernetes-version.md [hunk 13]
```diff
@@ -0,0 +1,159 @@
...
+Then repeat the above process one minor version at a time.
+
+For more information about EKS upgrades, please refer to the [EKS documentation](https://docs.aws.amazon.com/eks/latest/userguide/update-cluster.html).
+
+</TabItem>
+
+
+<TabItem label="Azure" value="azure">
+
+The Azure Kubernetes Service (AKS) requires that you specify the major, minor and patch version you wouldl like to use. To specify a Kubernetes version update the `azure.kubernetes_version` to `1.26.7` (or the version you need to upgrade to). Then run `nebari deploy` to apply these changes; you will get a validation error if you try to select a Kubernetes version that is unsupported by AKS or a version higher than `HIGHEST_SUPPORTED_K8S_VERSION`. This deployment process might take as long as 30 minutes.
```
Comment (pavithraes) [inline docs/docs/how-tos/upgrade-kubernetes-version.md:114]:
```suggestion
The Azure Kubernetes Service (AKS) requires that you specify the major, minor, and patch version you would like to use. To specify a Kubernetes version update the `azure.kubernetes_version` to `1.26.7` (or the version you need to upgrade to). Then run `nebari deploy` to apply these changes. This deployment process might take as long as 30 minutes.
```

File: docs/docs/how-tos/upgrade-kubernetes-version.md [hunk 14]
```diff
@@ -0,0 +1,159 @@
...
+
+:::info
+You will get a validation error if you try to select a Kubernetes version that is unsupported by GKE or a version higher than [`HIGHEST_SUPPORTED_K8S_VERSION`](https://github.com/nebari-dev/nebari/blob/91792952b67074b5c15c3b4009bde5926ca4ec6b/src/_nebari/constants.py#L11).
```
Comment (pavithraes) [inline docs/docs/how-tos/upgrade-kubernetes-version.md:117]:
```suggestion
You will get a validation error if you try to select a Kubernetes version that is unsupported by AKS or a version higher than [`HIGHEST_SUPPORTED_K8S_VERSION`][highest-supported-k8s].
```

File: docs/docs/how-tos/upgrade-kubernetes-version.md [hunk 15]
```diff
@@ -0,0 +1,159 @@
...
+Then repeat the above process one minor version at a time.
+
+For more information about DOKS upgrade, please refer to the [DOKS documentation](https://docs.digitalocean.com/products/kubernetes/how-to/upgrade-cluster/).
+
+
+</TabItem>
+
+
+</Tabs>
+
```
Comment (pavithraes) [inline docs/docs/how-tos/upgrade-kubernetes-version.md:159]:
```suggestion

<!-- Reusable links -->

[highest-supported-k8s]: https://github.com/nebari-dev/nebari/blob/91792952b67074b5c15c3b4009bde5926ca4ec6b/src/_nebari/constants.py#L11

```

File: docs/docs/how-tos/upgrade-kubernetes-version.md [hunk 16]
```diff
@@ -0,0 +1,159 @@
...
+1.28.2-do.0     1.28.2                cluster-autoscaler, docr-integration, ha-control-plane, token-authentication
+1.27.6-do.0     1.27.6                cluster-autoscaler, docr-integration, ha-control-plane, token-authentication
+1.26.9-do.0     1.26.9                cluster-autoscaler, docr-integration, ha-control-plane, token-authentication
+1.25.14-do.0    1.25.14               cluster-autoscaler, docr-integration, ha-control-plane, token-authentication
+```
+
+To upgrade your DOKS cluster, update the `digital_ocean.kubernetes_version` field in your `nebari-config.yaml` to match one of these DOKS Kubernetes versions. Then run `nebari deploy` to apply these changes. This deployment process might take as long as 30 minutes.
+
+:::info
+You will get a validation error if you try to select a Kubernetes version that is unsupported by DOKS or a version higher than [`HIGHEST_SUPPORTED_K8S_VERSION`](https://github.com/nebari-dev/nebari/blob/91792952b67074b5c15c3b4009bde5926ca4ec6b/src/_nebari/constants.py#L11).
```
Comment (pavithraes) [inline docs/docs/how-tos/upgrade-kubernetes-version.md:147]:
```suggestion
You will get a validation error if you try to select a Kubernetes version that is unsupported by DOKS or a version higher than [`HIGHEST_SUPPORTED_K8S_VERSION`][highest-supported-k8s].
```

File: docs/docs/how-tos/upgrade-kubernetes-version.md [hunk 17]
```diff
@@ -0,0 +1,159 @@
...
+
+</TabItem>
+
+
+<TabItem label="AWS" value="aws">
+
+The AWS Elastic Kubernetes Service (EKS) only requires that you supply the major and minor version of Kubernetes that you want. To specify Kubernetes version `1.26.7`, simply updated the `amazon_web_services.kubernetes_version` to `1.26`. Then run `nebari deploy` to apply these changes; you will get a validation error if you try to select a Kubernetes version that is unsupported by EKS or a version higher than `HIGHEST_SUPPORTED_K8S_VERSION`. This deployment process might take as long as 30 minutes.
+
+:::info
+You will get a validation error if you try to select a Kubernetes version that is unsupported by EKS or a version higher than [`HIGHEST_SUPPORTED_K8S_VERSION`](https://github.com/nebari-dev/nebari/blob/91792952b67074b5c15c3b4009bde5926ca4ec6b/src/_nebari/constants.py#L11).
```
Comment (pavithraes) [inline docs/docs/how-tos/upgrade-kubernetes-version.md:98]:
```suggestion
You will get a validation error if you try to select a Kubernetes version that is unsupported by EKS or a version higher than [`HIGHEST_SUPPORTED_K8S_VERSION`][highest-supported-k8s].
```

File: docs/docs/how-tos/upgrade-kubernetes-version.md [hunk 18]
```diff
@@ -0,0 +1,159 @@
...
+  ...
+  - 1.24.15-gke.1700
+  - 1.24.14-gke.2700
+```
+
+
+To upgrade your GKE cluster, update the `google_cloud_platform.kubernetes_version` field in your `nebari-config.yaml` to match one of these GKE Kubernetes versions. Then run `nebari deploy` to apply these changes. This deployment process might take as long as 30 minutes.
+
+:::info
+You will get a validation error if you try to select a Kubernetes version that is unsupported by GKE or a version higher than [`HIGHEST_SUPPORTED_K8S_VERSION`](https://github.com/nebari-dev/nebari/blob/91792952b67074b5c15c3b4009bde5926ca4ec6b/src/_nebari/constants.py#L11).
```
Comment (pavithraes) [inline docs/docs/how-tos/upgrade-kubernetes-version.md:79]:
```suggestion
You will get a validation error if you try to select a Kubernetes version that is unsupported by GKE or a version higher than [`HIGHEST_SUPPORTED_K8S_VERSION`][highest-supported-k8s].
```

**Review by pavithraes — APPROVED** [review]
Thanks, Eskild!

Most suggestions are related to wording, so I'll approve and you can self merge when ready. :)

**Pull request merged**